Fix bug breaking nested status fields

Kidswiss opened the pull request:
There's currently a fix in terrajet main branche that fixes a bug with
nested status fields. Nested status fields weren't handled correctly
before.

That bug made the actual S3 credentials appear in the status field of
the cluster scoped ManagedResource object of terrajet. They are now
correctly put into a connection secret, without manually mapping them.

Reference: https://github.com/crossplane/terrajet/issues/27

<!--
Thank you for helping to improve Crossplane!

Please read through https://git.io/fj2m9 if this is your first time opening a
Crossplane pull request. Find us in https://slack.crossplane.io/messages/dev if
you need any help contributing.
-->

### Description of your changes

<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open Crossplane issue. If yours does, you
can uncomment the below line to indicate which issue your PR fixes, for example
"Fixes #500":

-->
Fixes -

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

I've run the provider with a token, created an objects user and verified that the secret is written correctly.

Kidswiss committed:
Fix bug breaking nested status fields

There's currently a fix in terrajet main branche that fixes a bug with
nested status fields. Nested status fields weren't handled correctly
before.

That bug made the actual S3 credentials appear in the status field of
the cluster scoped ManagedResource object of terrajet. They are now
correctly put into a connection secret, without manually mapping them.

Reference: https://github.com/crossplane/terrajet/issues/27

ccremer reviewed (approved):
I can't really judge, I haven't had a look at the rest of the codebase.

Kidswiss commented:
> I can't really judge, I haven't had a look at the rest of the codebase.

Well the only manual change in the code i in `config.go` all the rest is generated anyway.

ccremer commented:
> Well the only manual change in the code i in `config.go` all the rest is generated anyway.

I'm aware, but it's exactly that I cannot judge :)

zugao reviewed (approved):
Seems ok with the non generated go code. Approved.

Kidswiss commented:
They've already released a new version of the terraform provider with the secret fixes. This makes parts of this PR already obsolete. Will adjust the changes accordingly.

Kidswiss committed:
Update to cloudscale 4.0 terraform

Kidswiss committed:
Allow terraform to update versions during init

The terraform lock file is cached between github CI runs. In order for
terraform to allow a newer version of a given module, we need to pass
an additional flag.

Kidswiss commented on Makefile:
```diff
@@ -92,7 +92,7 @@ $(TERRAFORM_PROVIDER_SCHEMA): $(TERRAFORM)
 	@mkdir -p $(TERRAFORM_WORKDIR)
 	@echo '{"terraform":[{"required_providers":[{"provider":{"source":"'"$(TERRAFORM_PROVIDER_SOURCE)"'","version":"'"$(TERRAFORM_PROVIDER_VERSION)"'"}}],"required_version":"'"$(TERRAFORM_VERSION)"'"}]}' > $(TERRAFORM_WORKDIR)/main.tf.json
-	@$(TERRAFORM) -chdir=$(TERRAFORM_WORKDIR) init > $(TERRAFORM_WORKDIR)/terraform-logs.txt 2>&1
+	@$(TERRAFORM) -chdir=$(TERRAFORM_WORKDIR) init -upgrade > $(TERRAFORM_WORKDIR)/terraform-logs.txt 2>&1
```
@ccremer how do you usually handle output in Makefiles?

Print everything? Only print stderr?

ccremer replied:
depends on the relevance and maybe also length, but I'm not really consistent either. when there's a `make` target for printing `helm template` as a form of manual verification, then obviously everything. `mkdir`'s,  `rm` or `sed`s not.
If the output is, I dunno, 20 lines but it's quite important because you run it regularly, then I'd display all output.
Basically, if you think it's worth printing everything, go ahead, otherwise logfiles are fine too.

Also worth including in that decision is GitHub actions. If something fails, it's gonna be hard to print from logfiles in a GitHub action run if the stderr doesn't reveal the actual error... (e.g. renovate etc)

Kidswiss replied:
Thanks for the explanation. If there's no consistent way I'll leave it this way for now.
